### PR TITLE
Added UI for read receipts in message info dialog

### PR DESCRIPTION
**Pull request by meganoahj**
ReadReceiptBox inspired by WhatsApp iOS

### CODE REVIEW

File: src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+export function ReadReceiptsList(props: ReadReceiptsListProps) {
+  const tx = useTranslationFunction()
+  const accountId = selectedAccountId()
+  const [receipts, setReceipts] = useState<T.MessageReadReceipt[]>([])
+  useEffect(() => {
+    BackendRemote.rpc
+      .getMessageReadReceipts(accountId, props.messageId)
+      .then(setReceipts)
+  }, [props.messageId, accountId])
```
Comment (WofWca) [inline src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx:21]: Is this info not available in the parent component?

File: themes/_themebase.scss [hunk 2]
```diff
@@ -217,9 +217,14 @@ $hover-contrast-change: 2%;
   --delta-btn-tab-selected-bg: #{$colorPrimary};
   --delta-btn-tab-selected-text: #{$textPrimary};
 
+  /** MessageDetails Colors **/
+  --MessageDetails-ReadReceipt-Bg: #{changeContrast($bgPrimary, 10%)};
+  --MessageDetails-ReadReceipt-Date: #{changeContrast($textPrimary, 30%)};
```
Comment (WofWca) [inline themes/_themebase.scss:222]: IDK about introducing these two variables on a theme-level. Sounds overly specific.

File: src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  if (receipts.length === 0) {
+    return null
+  }
+  return (
+    <div className={styles.ReadReceiptsContainer}>
+      <div className={styles.Heading}>
+        <div className={styles.DoubleCheckmarkIcon}></div>
+        {tx('pref_read_receipts')}
```
Comment (WofWca) [inline src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx:30]: I think we'll need a new string.

File: src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      </div>
+      <div className={styles.ReadReceiptBox}>
+        {receipts.map(receipt => (
+          <ReadReceipt receipt={receipt} />
+        ))}
+      </div>
```
Comment (WofWca) [inline src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx:36]: `<ol>` with `<li>` is better semantically, e.g. for screen readers.

**Review by WofWca — APPROVED** [review]

File: src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx (hunk 1 above)
Comment (WofWca) [inline src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx:21]: Also this doesn't react to new read receipts after render? Would it be hard to add?

File: src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx (hunk 4 above)
Comment (WofWca) [reply src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx:36]: Or a `<table>` even

File: themes/_themebase.scss (hunk 2 above)
Comment (Simon-Laux) [reply themes/_themebase.scss:222]: let's better discuss this practice in https://github.com/deltachat/deltachat-desktop/issues/3018

I think overly specific vars are better than say reusing chatlist hover color somewhere in an unrelated component.

File: src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx (hunk 1 above)
Comment (Simon-Laux) [reply src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx:21]: the dialog could add an event listener for read message that compares the message id and calls a refresh function. I can do a meeting with @meganoahj to teach him how.

Comment (meganoahj) [reply src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx:21]: no

File: _locales/_untranslated_en.json [hunk 5]
```diff
@@ -62,5 +62,8 @@
   },
   "search_in": {
     "message": "Search in %1$s"
+  },
+  "pref_read_by": {
```
Comment (WofWca) [inline _locales/_untranslated_en.json:66]: Doesn't `pref` mean "preference", i.e. "settings"?

File: src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    <div className={styles.ReadReceiptsContainer}>
+      <div className={styles.Heading}>
+        <div className={styles.DoubleCheckmarkIcon}></div>
+        {tx('pref_read_by')}
+      </div>
+      <table className={styles.ReadReceiptBox}>
+        {receipts.map(receipt => (
+          <ReadReceipt receipt={receipt} />
+        ))}
+      </table>
```
Comment (WofWca) [inline src/renderer/components/dialogs/MessageDetail/ReadReceipts.tsx:44]: This is not a valid table though, it doesn't have rows and columns just `<div>`s. If you don't want to get too deep, `<ul>` with `<li>`s would also be fine.

**Review by Simon-Laux — APPROVED** [review]

**Pull request merged**